### PR TITLE
Fix usage of PEM file

### PR DESCRIPTION
**Pull request by isarphilipp**
Closes https://github.com/codegreencreative/laravel-samlidp/issues/28

### CODE REVIEW

File: src/Jobs/SamlSso.php [hunk 1]
```diff
@@ -38,6 +39,10 @@ class SamlSso implements SamlContract
 {
     use Dispatchable, PerformsSingleSignOn;
 
+    protected $authn_request;
```
Comment (isarphilipp) [inline src/Jobs/SamlSso.php:42]: Those two properties have not been defined yet, but used below.

File: src/Jobs/SamlSso.php [hunk 2]
```diff
@@ -122,7 +127,7 @@ public function response()
 
             $encryptedAssertion = new EncryptedAssertionWriter();
             $encryptedAssertion->encrypt($assertion, KeyHelper::createPublicKey(
-                (new X509Certificate)->loadPem($this->sp_certificate)
+                $this->certificate
```
Comment (isarphilipp) [inline src/Jobs/SamlSso.php:130]: Pushing the loading of the certificate down to the `setSpCertificate()` method

File: src/Jobs/SamlSso.php [hunk 3]
```diff
@@ -203,9 +208,11 @@ private function getQueryParams()
...
     public function setSpCertificate()
     {
-        $this->sp_certificate = config(sprintf(
+        $filename = config(sprintf(
             'samlidp.sp.%s.certificate',
             $this->getServiceProvider($this->authn_request)
-        ));
+        ), 'cert.pem');
+
+        $this->certificate = (new X509Certificate)->loadPem(Storage::disk('samlidp')->get($filename));
```
Comment (isarphilipp) [inline src/Jobs/SamlSso.php:216]: This is the place where the new certificate should be loaded. 
There's no reason to use a `$this->sp_certificate` variable when we can directly load the certificate into `$this->certificate`.

**Comment by abublihi** [comment]
@isarphilipp I think you have to review the [PerformsSingleSignOn ](https://github.com/codegreencreative/laravel-samlidp/blob/516db46d28b422a33cdba3fcfde7260c0d5e6205/src/Traits/PerformsSingleSignOn.php#L12) trait.

File: src/Jobs/SamlSso.php (hunk 3 above)
Comment (abublihi) [reply src/Jobs/SamlSso.php:216]: `$this->certificate` is used for loading the Identity Provider certificate and you are trying to load the Service Provider using the same prop.

File: src/Jobs/SamlSso.php [hunk 4]
```diff
@@ -203,9 +208,11 @@ private function getQueryParams()
 
     public function setSpCertificate()
     {
-        $this->sp_certificate = config(sprintf(
+        $filename = config(sprintf(
             'samlidp.sp.%s.certificate',
             $this->getServiceProvider($this->authn_request)
-        ));
+        ), 'cert.pem');
```
Comment (abublihi) [inline src/Jobs/SamlSso.php:214]: If the certificate not found you will use the IDP certificate, why?

**Comment by upwebdesign** [comment]
This PR is stale and has not been addressed in a long time.

**Pull request closed without merging**